src/partition.mjs

```diff
@@ -1,30 +1,39 @@
 import Dequeue from 'dequeue'
-import iterable from './iterable'
+import ensureIterable from './internal/ensure-iterable'
 
-function uncurried (func, iter) {
+function partition (func, iter) {
   const satisfied = new Dequeue()
   const unsatisfied = new Dequeue()
-  const iterator = iterable(iter)[Symbol.iterator]()
+  const iterator = ensureIterable(iter)[Symbol.iterator]()
+  let exhausted = 0
 
   function * part (queue) {
-    while (true) {
-      while (queue.length) {
-        yield queue.shift()
-      }
+    try {
+      while (true) {
+        while (queue.length) {
+          yield queue.shift()
+        }
 
-      const { value, done } = iterator.next()
-      if (done) break
+        const { value, done } = iterator.next()
+        if (done) break
 
-      const chosen = func(value) ? satisfied : unsatisfied
-      chosen.push(value)
+        const chosen = func(value) ? satisfied : unsatisfied
+        chosen.push(value)
+      }
+    } finally {
+      exhausted++
+      if (exhausted === 2) {
+        if (typeof iterator.return === 'function') iterator.return()
+      }
```

Comment:
I don't understand, what is this for?

Reply:
Nevermind, I shoud've read [this](https://github.com/sithmel/iter-tools/wiki/Generator-functions-work-flow), my apologies.

Reply:
iterator.return is only useful to close an iterator **if is not exhausted**

Reply:
More info here: http://exploringjs.com/es6/ch_iteration.html#_closing-iterators-via-return

Reply:
Man! This iterator thing is more complicated than meets the eyes. Anyway, is there a way to make a wrapper that calls `iter.return()`? I prefer a clean and uncluttered code.

Reply:
I don't think there is a neater way. The good news is that is only necessary when you are consuming an iterator manually using "next". In any other case this is automatic.

Reply:
> I don't think there is a neater way. 

What about https://github.com/sithmel/iter-tools/pull/129?

I can go even further, make a wrapper that also support `tee()`.